### DISCUSSION TITLE
Realsing and pushing Homebrew package to GitLab projects in deeply nested groups

### DISCUSSION THREAD

**Discussion by jacobk** [jreleaser/jreleaser#1620] [General]
I have/need the following setup:
A. Project A (Java project, using gradle, producing a distribution. A Cli, JAVA-BINARY). It's hosted in a deeply nested group on gitlab.com. For example **gitlab.com/acme/foo/bar/ProjectA**
B. A dedicated project used as a homebrew tap. For example **gitlab.com/acme/foo/bar/homebrew-tap**

What assumption does jreleaser do about the relationship of the above?

If I try to configure jreleaser to use repo owner+user combo using either
* Project Access token (Created two tokens, one for A. and one for B)
* Group access token (token created at the *bar* level)
* Personal Access Token (based on my user)

Jreleaser fails the release with 404. I think is caused by that jreleaser seems to
1. Get current user (from access token)
2. Find projects based on the user. E.g. `https://gitlab.com/api/v4/user/{userId}/projects`. The issue is that this returns an empty list (seems to be described in GitLab API docs https://docs.gitlab.com/ee/api/projects.html#list-user-projects)
> Only the projects in the user’s (specified in user_id) namespace are returned. Projects owned by the user in any group or subgroups are not returned. An empty list is returned if a profile is set to private.

If I configure Jreleaser with `release` with an explicit Project Identifier the release works!
However, this seems to confuse the Homebrew packager. It continues to push the brew Formual + overriding the README in "Project A". From my digging there seems to be some assumption that the brew packager just reuses the same repo-config as the releaser?

It doesn't matter that I've explicitly configured a different owner/user for the brew packager.

The trace from jreleaser looks like this

```
[INFO]      [brew] setting up repository acme/foo/bar/homebrew-tap
[DEBUG]     [brew] locating repository acme/foo/bar/homebrew-tap
[DEBUG]     [brew] lookup repository acme/foo/bar/homebrew-tap
[DEBUG]     [brew] fetching project with GitLab id XXXXXXX # <- This is ProjectA'a id configured from release.gitlabl.projectIdentifier
[DEBUG]     [brew] found Acme / foo / bar / ProjectA (ID: XXXXXXX)
[DEBUG]     [brew] cloning https://gitlab.com/acme/foo/bar/ProjectA.git
```

Notice the switch at line 4. It logs that it's working towards the repo I configured in the brew packager, but then proceeds to use the project identifier from the release.

What seems to be missing is the option to specify `projectIdentifier` for the brew packager. From what I understand the gitlabl API doesn't seem to support discovering repositories inside groups based on user identity.

**Comment by aalmiray** [comment]
Missing an option to specify a `projectIdentifier` for brew's repository is indeed a problem that must be addressed. 

Moreover, the assumption that the release service is the same as the packager's may need to be revised. Someone may want to release distributions to GitLab while publishing some package manager files to GitHub. 

Could you file a ticket for the first issue? Will get it fixed by the next release 😅

**Reply by jacobk** [reply]
Thanks! I've created a ticket: https://github.com/jreleaser/jreleaser/issues/1621
